sa/storage-authority.go

```diff
@@ -253,20 +235,25 @@ func (ssa *SQLStorageAuthority) GetValidAuthorizations(ctx context.Context, regi
 		AND expires > ?
 		AND identifier IN (`+strings.Join(qmarks, ",")+`)
 		AND status = 'valid'
-		ORDER BY expires ASC
 		`, append([]interface{}{registrationID, now}, params...)...)
 	if err != nil {
 		return nil, err
 	}
 
 	byName := make(map[string]*core.Authorization)
 	for _, auth := range auths {
+		// No real life authorizations should have a nil expires. If we find them,
```

Comment:
Technically it's faster here: O(n) to scan and pick latest, vs O(n log n) on the database to sort. I do agree we should have metrics in SA, though. Filed: https://github.com/letsencrypt/boulder/issues/1791